### PR TITLE
Fix func signature for void return throws

### PR DESCRIPTION
**Pull request by kdawgwilk**
AutoMockable wasn't generating valid code for requirements that were void return throwing.  This PR adds an else for void returning methods and checks if throwing and adds throws if needed.  I have testing this in my project is now generating code that compiles

### CODE REVIEW

File: Templates/Templates/AutoMockable.stencil [hunk 1]
```diff
@@ -77,7 +77,7 @@ import AppKit
         {% call methodReceivedParameters method %}
     }
 {% else %}
-    func {{ method.name }}{% if not method.returnTypeName.isVoid %} {% if method.throws %}throws {% endif %}-> {{ method.returnTypeName }}{% endif %} {
```
Comment (ilyapuchka) [inline Templates/Templates/AutoMockable.stencil]: I think we can actually rewrite it like this:

```
func {{ method.name }}{% if method.throws %} throws {% endif %}{% if not method.returnTypeName.isVoid %} -> {{ method.returnTypeName }}{% endif %} {
```

**Comment by ilyapuchka** [comment]
@kdawgwilk thanks for the fix, I came up with a simpler suggestion though, what do you think?
Also do not forget to update CHANGELOG

@krzysztofzablocki @Antondomashnev @Liquidsoul is it possible with dagger to check for changes in non-swift files? I think we should extend this to templates.

**Comment by krzysztofzablocki** [comment]
@ilyapuchka sure, its just doing grep on Sourcery folder, you can add the templates folder for changes check

**Comment by Liquidsoul** [comment]
@ilyapuchka @krzysztofzablocki I've added the `Templates` folder for changelog warnings in Danger here #484

**Comment by krzysztofzablocki** [comment]
@kdawgwilk could you apply @ilyapuchka comment to template code before merging?

**Comment by kdawgwilk** [comment]
Sorry it took me so long to get back to this but thanks @ilyapuchka for the other PR fixing this issue 👍

**Pull request closed without merging**